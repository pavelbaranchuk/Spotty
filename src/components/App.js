import React, { Component } from "react";
import { Router, Switch, Redirect } from "react-router-dom";

import { withStyles } from "@material-ui/core/styles";
import PropTypes from "prop-types";

import ThemeProvider from "./shared/ThemeProvider";
import MySongs from "./mySongs/MySongs";
import Artists from "./artists/Artists";
import About from "./about/About";
import Albums from "./albums/Albums";
import ShareView from "./shareView/ShareView";
import AlbumTable from "./albums/AlbumTable";
import ArtistTable from "./artists/ArtistTable";
import NotFound from "./notFound/NotFound";
import Auth from "./auth/Auth";

import { connect } from "react-redux";
import { fetchUserAndTheme } from "../store/actionCreators/auth";
import PrivateRoute from "./routes/PrivateRoute";
import PublicRoute from "./routes/PublicRoute";
import history from "../history";

const styles = () => ({
  root: {
    display: "flex"
  }
});

export class App extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    fetchUserAndTheme: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.fetchUserAndTheme();
  }

  render() {
    const { classes } = this.props;

    return (
      <ThemeProvider>
        <Router history={history}>
          <div className={classes.root}>
            <Switch>
              <PublicRoute exact path="/login" component={Auth} />
              <PublicRoute exact path="/songs/:id" component={ShareView} />
              <Redirect exact from="/" to="/albums" />
              <PrivateRoute exact path="/mysongs" component={MySongs} />
              <PrivateRoute exact path="/albums" component={Albums} />
              <PrivateRoute exact path="/artists" component={Artists} />
              <PrivateRoute path="/about" component={About} />
              <PrivateRoute path="/albums/:id" component={AlbumTable} />
              <PrivateRoute path="/artists/:id" component={ArtistTable} />
              <PrivateRoute component={NotFound} />
            </Switch>
          </div>
        </Router>
      </ThemeProvider>
    );
  }
}

const mapDispatchToProps = {
  fetchUserAndTheme
};

export const ThemedApp = withStyles(styles)(App);
export default connect(
  null,
  mapDispatchToProps
)(ThemedApp);
