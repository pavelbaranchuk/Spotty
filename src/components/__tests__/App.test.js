import React from "react";
import { ThemedApp } from "../App";
import renderer from "react-test-renderer";

jest.mock("../shared/ThemeProvider", () => "theme-provider");
jest.mock("../shareView/ShareView", () => "share-view");
jest.mock("../mySongs/MySongs", () => "my-songs");
jest.mock("../artists/Artists", () => "artists");
jest.mock("../about/About", () => "about");
jest.mock("../albums/Albums", () => "albums");
jest.mock("../albums/AlbumTable", () => "album-table");
jest.mock("../artists/ArtistTable", () => "artist-table");
jest.mock("../auth/Auth", () => "auth");
jest.mock("../notFound/NotFound", () => "not-found");

jest.mock("../routes/PrivateRoute", () => "private-route");
jest.mock("../routes/PublicRoute", () => "public-route");

jest.mock("../../services/FirebaseService");

describe("App component", () => {
  const props = {
    fetchUserAndTheme: jest.fn()
  };

  it("should render correctly", () => {
    const tree = renderer.create(<ThemedApp {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should load initial data after mount", () => {
    expect(props.fetchUserAndTheme).toHaveBeenCalled();
  });
});
