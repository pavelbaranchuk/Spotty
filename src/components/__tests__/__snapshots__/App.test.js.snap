// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<theme-provider>
  <router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <div
      className="App-root-1"
    >
      <router-switch>
        <public-route
          component="auth"
          exact={true}
          path="/login"
        />
        <public-route
          component="share-view"
          exact={true}
          path="/songs/:id"
        />
        <router-redirect
          exact={true}
          from="/"
          to="/albums"
        />
        <private-route
          component="my-songs"
          exact={true}
          path="/mysongs"
        />
        <private-route
          component="albums"
          exact={true}
          path="/albums"
        />
        <private-route
          component="artists"
          exact={true}
          path="/artists"
        />
        <private-route
          component="about"
          path="/about"
        />
        <private-route
          component="album-table"
          path="/albums/:id"
        />
        <private-route
          component="artist-table"
          path="/artists/:id"
        />
        <private-route
          component="not-found"
        />
      </router-switch>
    </div>
  </router>
</theme-provider>
`;
